### PR TITLE
Update Prepop branch with the latest changes from main

### PR DESCRIPTION
**Pull request by kylelawsonAND**
### What is the context of this PR?
Updating the Prepop feature branch with the latest changes from main. 
### How to review
Check all tests are passing and main changes are fully reflected.

### CODE REVIEW

File: app/questionnaire/questionnaire_schema.py [hunk 1]
```diff
@@ -1271,6 +1271,10 @@ def is_summary_with_calculation(summary_type: str) -> bool:
     return summary_type in {"GrandCalculatedSummary", "CalculatedSummary"}
 
 
+def is_summary_with_calculation(summary_type: str) -> bool:
```
Comment (katie-gardner) [inline app/questionnaire/questionnaire_schema.py:1274]: this got duplicated during merge

**Review by katie-gardner — CHANGES_REQUESTED** [review]

File: app/questionnaire/questionnaire_schema.py (hunk 1 above)
Comment (kylelawsonAND) [reply app/questionnaire/questionnaire_schema.py]: Removed 👍

File: app/questionnaire/questionnaire_schema.py [hunk 2]
```diff
@@ -1109,7 +1113,9 @@ def _get_section_and_block_ids_dependencies_for_progress_source_and_answer_ids_f
                     dependencies_ids_for_progress_value_source["blocks"][section_id] = {
                         identifier: OrderedSet()
```
Comment (katie-gardner) [inline app/questionnaire/questionnaire_schema.py:1114]: I know this is a merge and this was already there, but I think this should be changed to 
```
dependencies_ids_for_progress_value_source["blocks"][section_id] = {
                        identifier: OrderedSet([current_section_id])
                    }

Comment (katie-gardner) [reply app/questionnaire/questionnaire_schema.py:1114]: it would fix the problem you're having with `.append` (needs to be `add`)

Comment (kylelawsonAND) [reply app/questionnaire/questionnaire_schema.py:1114]: looks like the statement is duplicated, one with passing in the current_section_id and one not, will remove the one without

Comment (kylelawsonAND) [reply app/questionnaire/questionnaire_schema.py]: done 👍

**Pull request closed without merging**